src/univariate/continuous/johnson.jl

```diff
@@ -0,0 +1,141 @@
+"""
+    Johnson(ξ, λ, γ, δ)
+
+The Johnson's ``S_U``-distribution with parameters ξ, λ, γ and δ is a transformation of the normal distribution:
+
+```math
+z \\sim \\gamma + \\delta \\sinh^{-1}\\Big(\\frac{x-\\xi}{\\lambda}\\Big),
+```
+
+where ``z \\sim \\mathcal{N}(0,1)``.
+
+```julia
+Johnson()           # Equivalent to Johnson(0.0, 1.0, 0.0, 1.0)
+Johnson(ξ, λ, γ, δ) # Johnson's S_U-distrubtion with shape parameters ξ, λ, γ and δ
+
+params(d)           # Get the parameters, i.e. (ξ, λ, γ, δ)
+shape(d)            # Get the shape parameter, i.e. ξ
+scale(d)            # Get the scale parameter, i.e. λ
+```
+
+External links
+
+* [Johnson's ``S_U``-distribution on Wikipedia](http://en.wikipedia.org/wiki/Johnson%27s_SU-distribution)
+"""
+struct Johnson{T<:Real} <: ContinuousUnivariateDistribution
+	ξ::T
+	λ::T
+	γ::T
+	δ::T
+	Johnson{T}(ξ, λ, γ, δ) where {T} = new{T}(ξ, λ, γ, δ)
+end
+
+function Johnson(ξ::T, λ::T, γ::T, δ::T; check_args::Bool=true) where {T<:Real}
+    @check_args Johnson (λ, λ ≥ zero(λ)) (δ, δ ≥ zero(δ))
+	return Johnson{T}(ξ, λ, γ, δ)
+end
+
+Johnson() = Johnson{Float64}(0.0, 1.0, 0.0, 1.0)
+Johnson(ξ::Real, λ::Real, γ::Real, δ::Real; check_args::Bool=true) = Johnson(promote(ξ, λ, γ, δ)...; check_args=check_args)
+Johnson(ξ::Integer, λ::Integer, γ::Integer, δ::Integer; check_args::Bool=true) = Johnson(float(ξ), float(λ), float(γ), float(δ); check_args=check_args)
+
+@distr_support Johnson -Inf Inf
+
+#### Conversions
+
+convert(::Type{Johnson{T}}, ξ::S, λ::S, γ::S, δ::S) where {T<:Real, S<:Real} = Johnson(T(ξ), T(λ), T(γ), T(δ))
```

Comment:
I don't think this method (and similar ones for other distributions) is useful at all.
```suggestion
```

Reply:
Is it because functions shouldn't expect the distributions to be of type other than `Real`?

Reply:
I'll delete it anyway.

Reply:
No, it's because I don't think there should be (automatic) conversions of four real numbers to `Johnson` - the constructor exists for this purpose.